internal/backend/scheduler/scheduler.go

```diff
@@ -0,0 +1,120 @@
+package scheduler
+
+import (
+	"context"
+	"fmt"
+
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+	"go.uber.org/zap"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/temporalclient"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+const (
+	taskQueueName = "scheduler-task-queue"
+	workerID      = "scheduler-worker"
+	workflowName  = "scheduler_workflow"
+)
+
+type Scheduler struct {
+	temporal   temporalclient.Client
+	sl         *zap.SugaredLogger
+	dispatcher sdkservices.Dispatcher
+}
+
+func New(l *zap.Logger, tc temporalclient.Client) *Scheduler {
+	return &Scheduler{sl: l.Sugar(), temporal: tc}
+}
+
+func (sch *Scheduler) Start(ctx context.Context, dispatcher sdkservices.Dispatcher) error {
+	sch.dispatcher = dispatcher
+
+	w := worker.New(sch.temporal.Temporal(), taskQueueName, worker.Options{Identity: workerID})
+	w.RegisterWorkflowWithOptions(sch.workflow, workflow.RegisterOptions{Name: workflowName})
+
+	if err := w.Start(); err != nil {
+		return fmt.Errorf("schedule wf: worker start: %w", err)
+	}
+
+	return nil
+}
+
+func (sch *Scheduler) Create(ctx context.Context, tid sdktypes.TriggerID, schedule string) error {
+	l := sch.sl.With("trigger_id", tid.String())
+
+	_, err := sch.temporal.Temporal().ScheduleClient().Create(
```

Comment:
do we have issue with users in this case ? 
same as for delete

Reply:
and update! yeah, an attacker might do this and the error would come only after the schedule is created. looking into it.

Reply:
changed the order of temporal invocation. will happen after db, as the db guards that for now. not that i am happy about it.